firebase-firestore/src/commonMain/kotlin/dev/gitlive/firebase/firestore/GeoPointSerializer.kt

```diff
@@ -32,10 +32,10 @@ object FirebaseGeoPointSerializer : KSerializer<GeoPoint> {
     override fun deserialize(decoder: Decoder): GeoPoint {
         return if (decoder is FirebaseDecoder) {
             // special case if decoding. Firestore encodes and decodes GeoPoints without use of serializers
-            decoder.value as GeoPoint
+            GeoPoint(decoder.value as PlatformGeoPoint)
```

Comment:
Can we hit case when decoder.value is not PlatformGeoPoint (sub)class?

Reply:
yes. but that would mean that we are deserializing an incorrect data structure. eg you have a Timestamp but trying to desertialize a GeoPoint